### PR TITLE
UP-4860: Extract Bootstrap from uPortal, load from NPM package

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] view conforms with [WCAG 2.0 AA][]

##### Description of change

<https://issues.jasig.org/browse/UP-4860>

Extracts Bootstrap from uPortal source, instead compiles with version from NPM.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by MichaelVose2** [comment]
Hello,

This is definitely the direction to go in my opinion.

**Comment by ChristianMurphy** [comment]
@cousquer This works for standard/static skins as is.
Some additional work may need to be done to support dynamic respondr skins.

File: uPortal-webapp/src/main/webapp/media/skins/respondr/common/common.less [hunk 1]
```diff
@@ -17,17 +17,17 @@
...
   * Also, due to a limitation in the LESS CSS Maven Plugin, whitespace before an
   * @import will cause a compile error.
   */
- 
+
  /* ==========================================================================*
     Bootstrap
     ==========================================================================*/
-@import "bootstrap/bootstrap.less";
+@import "../../../../../../../node_modules/bootstrap/less/bootstrap.less";
```
Comment (bjagg) [inline uPortal-webapp/src/main/webapp/media/skins/respondr/common/common.less]: The magic!

Comment (drewwills) [inline uPortal-webapp/src/main/webapp/media/skins/respondr/common/common.less]: @bjagg -- Did you test this PR with `dynamic-respondr-skin`?

Comment (bjagg) [inline uPortal-webapp/src/main/webapp/media/skins/respondr/common/common.less]: @drewwills Correct me if I have this wrong. Dynamic respondr skins allows a selection of a skin available for the theme. The best practice to create these skins is to derived from defaultSkin, copying the directory and defaultSkin.less file. This file is where common/common.less is imported, which specifies Bootstrap's location.

It seems straightforward that derived skins following this practice would work.

I will be testing this shortly for some client work (but the merge should have waited.)

Comment (drewwills) [inline uPortal-webapp/src/main/webapp/media/skins/respondr/common/common.less]: The dymanic-respondr-skin portlet does allow you to select the skin upon which your skin will be based.  It also gives you several color-pickers that you can use to customize your "dynamic" skin.

When you hit Submit -- after using the color-pickers -- your color choices are used to recompile Bootstrap _from source_.  I haven't looked... but (with this update) are the Bootstrap LESS files (sources) present in Tomcat in the uPortal webapp?  Will they be available in the deployed/running portal for compiling via lessc?  And if they are, is everything wired up correctly -- paths, etc. -- for iti to work?

**Pull request merged**